File: TumeKyouen/Dao/TKSettingDao.m
//
//  TKSettingDao.m
//  TumeKyouen
//
//  Created by 石倉 昇 on 2013/07/14.
//  Copyright (c) 2013年 noboru. All rights reserved.
//

#import "TKSettingDao.h"

@implementation TKSettingDao

- (void)saveStageNo:(NSNumber *)stageNo {
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    [defaults setObject:stageNo forKey:@"stageNo"];
    [defaults synchronize];
}

- (NSNumber *)loadStageNo {
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    NSNumber *stageNo = [defaults objectForKey:@"stageNo"];
    if (stageNo == nil) {
        // デフォルトは1
        stageNo = @1;
    }
    return stageNo;
}

@end

File: TumeKyouen/Dao/TKSettingDao.swift
//
//  TKSettingDao.swift
//  TumeKyouen
//
//  Created by 石倉 昇 on 2016/02/03.
//  Copyright © 2016年 noboru. All rights reserved.
//

import Foundation

class TKSettingDao: NSObject {

    func saveStageNo(stageNo: NSNumber) {
        let defaults = NSUserDefaults.standardUserDefaults()
        defaults.setObject(stageNo, forKey: "stageNo")
        defaults.synchronize()
    }

    func loadStageNo() -> NSNumber {
        let defaults = NSUserDefaults.standardUserDefaults()
        if let stageNo = defaults.objectForKey("stageNo") as? NSNumber {
            return stageNo
        }

        return 1
    }
}
